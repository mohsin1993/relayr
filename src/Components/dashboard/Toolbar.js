import React from 'react';
import PropTypes from 'prop-types';

const Toolbar = ({ loadState, searchText, updateSearchText }) => (
  <div className='toolbar'>
    <button type='button' onClick={loadState}>
      Reload
    </button>
    <input
      type='text'
      placeholder='Search'
      value={searchText}
      onChange={e => updateSearchText(e.target.value)}
    />
  </div>
);

Toolbar.propTypes = {
  searchText: PropTypes.string,
  loadState: PropTypes.func.isRequired,
  updateSearchText: PropTypes.func.isRequired,
};

Toolbar.defaultProps = {
  searchText: '',
};

export default Toolbar;
