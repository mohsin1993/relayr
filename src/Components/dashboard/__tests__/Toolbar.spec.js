import React from 'react';
import snapshot from 'check-snapshot';
import { mount } from 'enzyme';
import Toolbar from '../Toolbar';

describe('Toolbar', () => {
  const loadStateMock = jest.fn();
  const updateSearchText = jest.fn();

  it('Render toolbar correctly', () => {
    snapshot(
      <Toolbar
        searchText='acce'
        loadState={loadStateMock}
        updateSearchText={updateSearchText}
      />
    );
  });

  it('On click Triggers load device state func', () => {
    const wrapper = mount(
      <Toolbar loadState={loadStateMock} updateSearchText={updateSearchText} />
    );
    wrapper.find('button').simulate('click');
    expect(loadStateMock).toHaveBeenCalled();
  });

  it('On search input change Triggers updateSearchText with correct value', () => {
    const wrapper = mount(
      <Toolbar
        searchText=''
        loadState={loadStateMock}
        updateSearchText={updateSearchText}
      />
    );
    const newValue = 'Hello';
    wrapper.find('input').simulate('change', { target: { value: newValue } });
    expect(updateSearchText).toHaveBeenCalledWith(newValue);
  });
});
