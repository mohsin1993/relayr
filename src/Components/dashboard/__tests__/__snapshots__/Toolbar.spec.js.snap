// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Render toolbar correctly 1`] = `
<div
  className="toolbar"
>
  <button
    onClick={[MockFunction]}
    type="button"
  >
    Reload
  </button>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="acce"
  />
</div>
`;
